development-utility/development-server/src/com/thoughtworks/go/server/DevelopmentServer.java

```diff
@@ -96,10 +98,8 @@ private static void copyDbFiles() throws IOException {
     }
 
     private static void copyActivatorJarToClassPath() throws IOException {
-        File activatorJarFromTarget = new File("../plugin-infra/go-plugin-activator/target/go-plugin-activator.jar");
-        File activatorJarFromMaven = new File(System.getProperty("user.home") + "/.m2/repository/com/thoughtworks/go/go-plugin-activator/1.0/go-plugin-activator-1.0.jar");
-        File activatorJar = activatorJarFromTarget.exists() ? activatorJarFromTarget : activatorJarFromMaven;
-        new SystemEnvironment().set(SystemEnvironment.PLUGIN_ACTIVATOR_JAR_PATH, "go-plugin-activator.jar");
+        File activatorJar = new File("../plugin-infra/go-plugin-activator/target/libs/").listFiles((FileFilter) new WildcardFileFilter("go-plugin-activator-*.jar"))[0];
+        new SystemEnvironment().set(SystemEnvironment.PLUGIN_ACTIVATOR_JAR_PATH, activatorJar.getName());
 
         if (activatorJar.exists()) {
             FileUtils.copyFileToDirectory(activatorJar, classpath());
```

Comment:
There is still something wrong.
When debugging I can see the jar copied to `target/classes`. But then [here](https://github.com/gocd/gocd/blob/master/plugin-infra/go-plugin-infra/src/com/thoughtworks/go/plugin/infra/listeners/DefaultPluginJarChangeListener.java#L209) activator is still not found. I don't understand why, `systemEnvironment.get(PLUGIN_ACTIVATOR_JAR_PATH)` evaluates to `go-plugin-activator-16.7.0-3722.jar`
But `getClass().getClassLoader().getResource(systemEnvironment.get(PLUGIN_ACTIVATOR_JAR_PATH));` evaluates to null.

Also bundled zip failing is causing the rest of plugin infra to not start. I had this change in `PluginsInitializer` to force starting plugins
```
@@ -61,11 +61,11 @@ public class PluginsInitializer implements Initializer {
                 FileUtils.cleanDirectory(bundledPluginsDirectory);
                 zipUtil.unzip(getPluginsZipStream(), bundledPluginsDirectory);
             }
-            pluginManager.startInfrastructure();
-            pluginManager.registerPluginsFolderChangeListener();
         } catch (Exception e) {
             LOG.error("Could not extract bundled plugins to default bundled directory", e);
         }
+        pluginManager.startInfrastructure();
+        pluginManager.registerPluginsFolderChangeListener();
     }
 
     ZipInputStream getPluginsZipStream() {
```